[DOCS] collection_release_with_branches.rst: update major releases section (#79335)
ansible/ansible: docs/docsite/rst/community/collection_contributors/collection_release_with_branches.rst

The new version is assumed to be ``X.0.0``.

> 1. Check the collections for deprecations that are planned for removal in the major release. Use past changelogs or run ``grep -r `X.0.0` plugins/`` in the repository.

Reviewer: This should only be done once the version in `galaxy.yml` has already been updated to X.0.0 and sanity tests are passing. That step should have made sure that most deprecations have already been handled, and this is an extra step to verify nothing was missed.

Author: @felixfontein sorry, I don't have a lot of experience with major releases, deprecations, etc. (i see the typo there btw it should be `collection`, not `collectionS`).
Could you please suggest your wording or elaborate a bit more?

Reviewer: I would probably make this 2., and add a 1. before it:
```rst
1. Make sure that ``galaxy.yml`` contains the correct version number ``X.0.0``. If that is not the case, create a PR to update it. This will make sanity tests fail for all deprecations that have to be removed in ``X.0.0``, so this is potentially a lot of work and should have been done weeks before the major release.
```
And then remove

> 6. In the ``stable-X`` branch, ensure that ``galaxy.yml`` contains the correct version number ``X.0.0``. If not, update it.

resp. change it to

> 6. In the ``stable-X`` branch, verify that ``galaxy.yml`` contains the correct version number ``X.0.0``.

Author: @felixfontein SGTM, thanks for clarifying, changed as you suggested, PTAL

> 1. Check the collections for deprecations that have been planned to be removed in the major release. Use past changelogs. Also running ``grep -r `X.0.0` plugins/`` in the repository can be helpful.

Reviewer 2: ```suggestion
1. Check the collections for deprecations that are planned for removal in the major release. Use past changelogs or run ``grep -r `X.0.0` plugins/`` in the repository.
```

Reviewer: should that be planned *for removal*, and not *for removed*?

Reviewer 2: yes, yes it should. what can I say? I guess my caffeine dependency wasn't fully loaded.

Author: LGTM, committed, thanks:)

> 2. If you are going to release the ``community.general`` and ``community.network`` collections, create new ``backport-X`` and ``needs_backport_to_stable_X`` labels in the corresponding repositories. Copy the styles and descriptions from the corresponding existing labels.

Reviewer: ```suggestion
2. If you are going to release the ``community.general`` and ``community.network`` collections, create a new ``backport-X`` label in the corresponding repositories. Copy the styles and descriptions from the corresponding existing labels.
```

The new version is assumed to be ``X.0.0``.

1. Make sure that ``galaxy.yml`` contains the correct version number ``X.0.0``. If that is not the case, create a PR to update it. This will make sanity tests fail for all deprecations that have to be removed in ``X.0.0``, so this is potentially a lot of work and should have been done weeks before the major release.

> 2. Check the collection for deprecations that are planned for removal in the major release. Use past changelogs or run ``grep -r `X.0.0` plugins/`` in the repository.

Reviewer: ```suggestion
2. Check the collection for deprecations that are planned for removal in the major release that were not reported by the sanity tests. Use past changelogs or run ``grep -r `X.0.0` plugins/`` in the repository.
```

Author: SGTM! Committed, PTAL
